src/python/gudhi/representations/metrics.py

```diff
@@ -10,17 +10,169 @@
+        kernel_approx: kernel approximation class used to speed up computation. Common kernel approximations classes can be found in the scikit-learn library (such as RBFSampler for instance).   
+
+    Returns: 
+        float: the persistence Fisher distance between persistence diagrams. 
+    """
+    projection = (1./2) * np.ones((2,2))
+    diagonal_projections1 = np.matmul(D1, projection)
+    diagonal_projections2 = np.matmul(D2, projection)
+    if kernel_approx is not None:
+        approx1 = kernel_approx.transform(D1)
+        approx_diagonal1 = kernel_approx.transform(diagonal_projections1)
+        approx2 = kernel_approx.transform(D2)
+        approx_diagonal2 = kernel_approx.transform(diagonal_projections2)
+        Z = np.concatenate([approx1, approx_diagonal1, approx2, approx_diagonal2], axis=0)
+        U, V = np.sum(np.concatenate([approx1, approx_diagonal2], axis=0), axis=0), np.sum(np.concatenate([approx2, approx_diagonal1], axis=0), axis=0) 
+        vectori, vectorj = np.abs(np.matmul(Z, U.T)), np.abs(np.matmul(Z, V.T))
+        vectori_sum, vectorj_sum = np.sum(vectori), np.sum(vectorj)
+        if vectori_sum != 0:
+            vectori = vectori/vectori_sum
+        if vectorj_sum != 0:
+            vectorj = vectorj/vectorj_sum
+        return np.arccos(  min(np.dot(np.sqrt(vectori), np.sqrt(vectorj)), 1.)  )
+    else:
+        Z = np.concatenate([D1, diagonal_projections1, D2, diagonal_projections2], axis=0)
+        U, V = np.concatenate([D1, diagonal_projections2], axis=0), np.concatenate([D2, diagonal_projections1], axis=0) 
+        vectori = np.sum(np.exp(-np.square(pairwise_distances(Z,U))/(2 * np.square(bandwidth)))/(bandwidth * np.sqrt(2*np.pi)), axis=1)
+        vectorj = np.sum(np.exp(-np.square(pairwise_distances(Z,V))/(2 * np.square(bandwidth)))/(bandwidth * np.sqrt(2*np.pi)), axis=1)
+        vectori_sum, vectorj_sum = np.sum(vectori), np.sum(vectorj)
+        if vectori_sum != 0:
+            vectori = vectori/vectori_sum
+        if vectorj_sum != 0:
+            vectorj = vectorj/vectorj_sum
+        return np.arccos(  min(np.dot(np.sqrt(vectori), np.sqrt(vectorj)), 1.)  )
+
+def sklearn_wrapper(metric, X, Y, **kwargs):
+    """
+    This function is a wrapper for any metric between two persistence diagrams that takes two numpy arrays of shapes (nx2) and (mx2) as arguments.
+    """
+    if Y is None:
+        def flat_metric(a, b):
+            return metric(X[int(a[0])], X[int(b[0])], **kwargs)
+    else:
+        def flat_metric(a, b):
+            return metric(X[int(a[0])], Y[int(b[0])], **kwargs)
+    return flat_metric
+
+PAIRWISE_DISTANCE_FUNCTIONS = {
+    "wasserstein": hera_wasserstein_distance,
+    "hera_wasserstein": hera_wasserstein_distance,
+    "persistence_fisher": persistence_fisher_distance,
+}
+
+def pairwise_persistence_diagram_distances(X, Y=None, metric="bottleneck", **kwargs):
```

Comment:
Replying to a point in the audio the other day: it would have been possible to have a class Pairwise_persistence_diagram_distances with fit() and transform(), if we wanted to use a single point of entry and still be usable in pipelines. Now I am not at all saying we should do that, I just felt like replying to that point.

Reply:
Indeed, that's possible. Although, such a class is not really necessary (it's my opinion of course) because:

---if you want to use sklearn pipelines, it's actually a little bit easier to cross-validate across multiple estimators than cross-validate across different values of a parameter in a single estimator.

---if you don't want to used sklearn pipelines, it is more intuitive to call `pairwise_persistence_diagram_distances()` than going through the fit and transform methods of estimators.